src/Support/SessionIdValidator.php

```
@@ -0,0 +1,104 @@
+     * Minimum recommended length for session IDs (for security).
+     */
+    public const MIN_RECOMMENDED_LENGTH = 16;
+
+    /**
+     * Check if a session ID is valid without throwing an exception.
+     *
+     * @param string $sessionId The session ID to validate
+     * @param bool $trim Whether to trim whitespace before validation (default: false)
+     * @return bool True if valid, false otherwise
+     */
+    public static function isValid(string $sessionId, bool $trim = false): bool
```

PHPMD (2.15.0)

The method isValid has a boolean flag argument $trim, which is a certain sign of a Single Responsibility Principle violation. (undefined)

(BooleanArgumentFlag)